iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -278,23 +419,33 @@ bool ProcessManager::removeProcess(const ProcessName_t& name) noexcept
     auto it = m_processList.begin();
     while (it != m_processList.end())
     {
-        if (it->getName() == name)
+        auto otherName = it->getName();
+        if (name == otherName)
         {
-            m_portManager.deletePortsOfProcess(name);
-
-            m_processIntrospection->removeProcess(it->getPid());
-
-            // delete application
-            it = m_processList.erase(it);
-
-            LogDebug() << "New Registration - removed existing application " << name;
+            if (removeProcess(it))
```

Comment:
This is a bug. If you iterate through a list and remove elements you have to increment the iterator first and then remove the element otherwise you have undefined behavior

Reply:
@elfenpiff are you sure it's a bug? The stackoverflow accepted answer mentions the following valid code to do this: `items.erase(i++);  // alternatively, i = items.erase(i);`

That's what `removeProcess(it)` does:
`processIter = m_processList.erase(processIter); // delete application`

Reply:
This is again not a bug. Hidden in `removeProcess(it)` there is a `it = list.erase(it);` like @marthtz mentioned. It's just not obvious from the caller side. Maybe this could be solved differently but I've currently no good idea how.

Reply:
I'm ready to dismiss it too! ;)